platform/linux_platform.go

```diff
@@ -1380,7 +1380,7 @@ func (p linux) MigratePersistentDisk(fromMountPoint, toMountPoint string) error
 
 	// Golang does not implement a file copy that would allow us to preserve dates...
 	// So we have to shell out to tar to perform the copy instead of delegating to the FileSystem
-	tarCopy := fmt.Sprintf("(tar -C %s --xattrs -cf - .) | (tar -C %s --xattrs -xpf -)", fromMountPoint, toMountPoint)
+	tarCopy := fmt.Sprintf("(tar -C %s --xattrs --xattrs-include=*.* -cf - .) | (tar -C %s --xattrs --xattrs-include=*.* -xpf -)", fromMountPoint, toMountPoint)
```

Comment:
This change breaks the following unit tests: 

`bosh-agent/platform/linux_platform_test.go:3704`
`bosh-agent/platform/linux_platform_test.go:3669`

Could you please fix them according to the new behavior and, as Ruben suggested, add a new one describing the expected behavior? For more info look [here](https://github.com/cloudfoundry/bosh-agent/blob/main/docs/dev_setup.md#running-tests).

Reply:
Thank's for spotting and explaining.
Fixed the tests:
```
Ginkgo ran 51 suites in 2m42.468308791s
Test Suite Passed

SUCCESS
```

Regarding the creation of a new test, I lack golang skills and deeper knowledge in this project, as I am not sure how to achieve this.
I would assume it would require an extension of the [FakeFileSystem](https://github.com/cloudfoundry/bosh-utils/blob/master/system/fakes/fake_file_system.go), as that one is not aware about extended attributes in general at the moment.
Is there already somewhere a test for the existing `--xattrs` flag, which I could somehow re-use and do things similar?

Reply:
@speer Looking at the code I see that creating a unit test for this might be a little hard because of the fake objects, the other route that could be taken is to create an integration test specific for linux, These kind of tests are designed to run on the target os (linux) , that could make things easier.
There is a _similar_ test that can be partially re used: `integration/ephemeral_disk_test.go`. 
@klakin-pivotal do you have any opinions on the testing (or not testing) of this feature?

Reply:
Given that the only tests I see for persistent disk migration are the tests in `migrate_disk_test.go` and `linux_platform_test.go` which use the `FakeFileSystem` class, it seems acceptable to me to not write an integration test for this and just update the `cmdRunner.RunCommands` expectations in the two tests that are failing because of the change to the `tar` invocation.

I had assumed that there was an integration test that tested Persistent Disk Migration, but now that I know that there's not one, I'm okay with just updating the existing tests.

I certainly wouldn't say "No" to a new integration test, and I agree that looking at `integration/ephemeral_disk_test.go` would be a reasonable way to begin to get an idea of how to create an integration test for this bug.

Reply:
> Given that the only tests I see for persistent disk migration are the tests in migrate_disk_test.go and linux_platform_test.go which use the FakeFileSystem class, it seems acceptable to me to not write an integration test for this and just update the cmdRunner.RunCommands expectations in the two tests that are failing because of the change to the tar invocation.

The two tests have been updated.
So would it be ok, if the new "persistent_disk" integration test would be created outside of this PR? I definitively lack knowledge in the project to be able to create such just by myself in reasonable time.

Reply:
Yeah, I think it can be done later. Thanks.